model/map/src/main/java/org/keycloak/models/map/serverinfo/MapServerInfoProviderFactory.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.map.serverinfo;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import org.jboss.logging.Logger;
+import org.keycloak.Config;
+import org.keycloak.common.Version;
+import org.keycloak.common.util.Base64Url;
+import org.keycloak.common.util.RandomString;
+import org.keycloak.migration.MigrationModel;
+import org.keycloak.migration.ModelVersion;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.models.ServerInfoProvider;
+import org.keycloak.models.ServerInfoProviderFactory;
+
+public class MapServerInfoProviderFactory implements ServerInfoProviderFactory {
+
+    @Override
+    public void init(Config.Scope config) {
+        String seed = config.get("resourcesVersionSeed");
+        if (seed == null) {
+            Logger.getLogger(ServerInfoProviderFactory.class).warn("It is recommended to set 'resourcesVersionSeed' property in the config especially in clustered environment. For more details see the documentation.");
+            //generate random string for this installation
+            seed = RandomString.randomCode(10);
+        }
+        try {
+            Version.RESOURCES_VERSION = Base64Url.encode(MessageDigest.getInstance("MD5")
+                    .digest((seed + new ModelVersion(Version.VERSION_KEYCLOAK).toString()).getBytes()))
+                    .substring(0, 5);
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void postInit(KeycloakSessionFactory factory) {
+    }
+
+    @Override
+    public ServerInfoProvider create(KeycloakSession session) {
+        return new ServerInfoProvider() {
```

Comment:
Can either `MapServerInfoProviderFactory` just also implement `ServerInfoProvider` and here return `this`, or even better, prepare this `ServerInfoProvider` as a `private static final` instance and return it?